xbmc/FileItem.cpp

```diff
@@ -2870,289 +2873,225 @@ void CFileItemList::Stack(bool stackFiles /* = true */)
-                  ++expr;
-                  break;
-                }
-              }
-              else if (!StringUtils::EqualsNoCase(Ignore1, Ignore2)) // False positive, try again with offset
-              {
-                offset = expr->GetSubStart(3);
-                break;
-              }
-              else // Extension mismatch
-              {
-                offset = 0;
-                ++expr;
-                break;
-              }
+              CFileItemPtr item{Get(index)};
+              item->SetPath(playPath); // updated path (for DVD/Bluray files)
+              item->m_bIsFolder = false;
+              item->SetLabel2("");
+              item->SetLabelPreformatted(true);
+              m_sortDescription.sortBy = SortByNone; // sorting is now broken
             }
-            else // Title mismatch
-            {
-              offset = 0;
-              ++expr;
-              break;
-            }
-          }
-          else // No match 2, next expression
-          {
-            offset = 0;
-            ++expr;
-            break;
+            if (stack.size() > 1)
+              deleteItems.emplace_back(index); // delete all but first
           }
-          j++;
         }
-        if (j == Size())
-          expr = stackRegExps.end();
-      }
-      else // No match 1
-      {
-        offset = 0;
-        ++expr;
-      }
-      if (stack.size() > 1)
-      {
-        // have a stack, remove the items and add the stacked item
-        // dont actually stack a multipart rar set, just remove all items but the first
-        std::string stackPath;
+
+        // Create stack
+        // Generate combined stack path
+        std::string stackPath{};
         if (Get(stack[0])->IsRAR())
           stackPath = Get(stack[0])->GetPath();
         else
-        {
-          CStackDirectory dir;
-          stackPath = dir.ConstructStackPath(*this, stack);
-        }
-        item1->SetPath(stackPath);
-        // clean up list
-        for (unsigned k = 1; k < stack.size(); k++)
-          Remove(i+1);
-        // item->m_bIsFolder = true;  // don't treat stacked files as folders
-        // the label may be in a different char set from the filename (eg over smb
-        // the label is converted from utf8, but the filename is not)
-        if (!CServiceBroker::GetSettingsComponent()->GetSettings()->GetBool(CSettings::SETTING_FILELISTS_SHOWEXTENSIONS))
+          stackPath = CStackDirectory::ConstructStackPath(*this, stack);
+
+        // First item in stack becomes the stack
+        std::string stackName = std::get<1>(stackBaseItem); // title
+        if (!CServiceBroker::GetSettingsComponent()->GetSettings()->GetBool(
+                CSettings::SETTING_FILELISTS_SHOWEXTENSIONS))
           URIUtils::RemoveExtension(stackName);
 
-        item1->SetLabel(stackName);
-        item1->m_dwSize = size;
-        break;
+        // Update item
+        CFileItemPtr stackItem{Get(stack[0])};
```

Comment:
While you are at it, and we want to get rid of `CFileItemPtr`, can you change this to `const auto stackItem{std::make_shared<CFileItem>(Get(stack[0])}; `?

Reply:
No, I think that's an oversight. Thanks.